content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/architecture.md

```diff
@@ -0,0 +1,143 @@
+  * Regions can be dynamically loaded and unloaded individually, making it easy to implement coarse-grained terrain streaming.
+* Each region can have different subsets of features.
+  * Regions where gameplay occurs can spawn terrain physics representations, but distant terrain regions that are for visuals only don't need to have physics enabled.
+  * Aspects of terrain rendering can be disabled for server-side or machine learning simulations that don't require a visual representation of terrain.
+* Different components can have different input data resolutions. The terrain system defines a single _output_ terrain grid and resolution, but the input data resolutions can be varied to optimize the data size and complexity based on the need for that region. Even within a region, two different data providers (such as height and surface type) can use two different resolutions based on what's appropriate for that type of data.
+* The functionality can be replaced at a component-by-component level.
+  * The PhysX Heightfield component could be replaced with one that connects to another physics system such as Jolt or Bullet.
+  * Image Gradients can be used for areas with authored input data, and Fast Noise Gradients can be used to procedurally generate data in other areas.
+  * New data providers can be created for streaming satellite data, performing complex procedural generation, etc, and swapped in without needing to replace the rest of the terrain system.
+
+In short, the singleton terrain system provides a single simplified API for runtime systems that need terrain data. The separated components provide better workflows and flexibility for content authors with multiple layers of data tuning and optimization.
+
+## Component design
+
+The specific functional and data separations in the terrain component design have been chosen to make it easy to control what functionality is active and to provide clear touchpoints for replacing functionality everywhere that it's likely to be useful.
+
+### Level components
+
+The terrain system uses three level components:
+
+| Component | Description |
+| - | - |
+| **Terrain World** | Controls the base terrain system. The entire system is enabled or disabled by adding or removing this component. The configuration parameters that affect the entire terrain world exist on this component as well.|
+| **Terrain World Renderer** | Controls the terrain rendering. It requires the **Terrain World** component to exist, and it provides all of the rendering configuration that needs to be consistent across the entire terrain. This component is kept separate from **Terrain World** so that it's possible to define a conceptual terrain without requiring the overhead of rendering for use cases where a visual representation simply isn't necessary. Also, this provides a "replacement touchpoint" for creating different terrain renderers, such as a voxel-based implementation or a raytracing-based implementation.|
```

Comment:
- For "...that needs to be...", the way I read this felt like there's emphasis on how the rendering _needs_ to be consistent, which then makes me wonder what that means and why it needs to be. Instead, I believe this sentence intends to indicate simply that the rendering configuration applies consistently throughout the entire terrain. If my assumption is right, see my suggestion below. 
- For the sentence "This component is kept separate...", broke up the sentence because it was getting too long to read easily.
```suggestion
| **Terrain World Renderer** | Controls the terrain rendering. It requires the **Terrain World** component to exist, and it provides all of the rendering configuration, which applies consistently across the entire terrain. This component is separated from **Terrain World** so that it's possible to define a conceptual terrain without requiring the overhead of rendering. This is useful for cases where a visual representation simply isn't necessary. Also, this provides a "replacement touchpoint" for creating different terrain renderers, such as a voxel-based implementation or a raytracing-based implementation.|
```

Reply:
I'll change the first one to "it provides all of the rendering configuration settings that apply globally to the entire terrain." The other part looks good to me.